### PR TITLE
Setup elf2dol

### PR DESCRIPTION
**Pull request by Yanis42**
This is technically not required to decompile but it is required if you want to build this project and use the result (also this was taken from [zeldaret/tp](https://github.com/zeldaret/tp), and using the elf directly doesn't work)

This PR also sets up ``NON_MATCHING`` if you want to make a mod, and improves things in the makefile (at least to me they're improvements)

I'm open to suggestions

---
To get a dol simply run ``make dol``, I kept the elf as the default target

### CODE REVIEW

**Comment by Yanis42** [comment]
gonna close this but I'll keep this branch, if someone needs it for some reasons feel free to take it

**Comment by Yanis42** [comment]
I pushed an update to this, the makefile has a comment above the ``dol`` rule to say it's meant for testing/modding

File: Makefile [hunk 1]
```diff
@@ -39,14 +42,19 @@ else
 	WINE := wine
 endif
 
-AS := $(DEVKITPPC)/bin/powerpc-eabi-as
-OBJCOPY := $(DEVKITPPC)/bin/powerpc-eabi-objcopy
-OBJDUMP := $(DEVKITPPC)/bin/powerpc-eabi-objdump
+PPC_BIN_PREFIX := $(DEVKITPPC)/bin/powerpc-eabi
```
Comment (Thar0) [inline Makefile:45]: The prefix should include the `-` to match how it works in other projects

File: Makefile [hunk 2]
```diff
@@ -39,14 +42,19 @@ else
...
+AS := $(PPC_BIN_PREFIX)-as
+OBJCOPY := $(PPC_BIN_PREFIX)-objcopy
+OBJDUMP := $(PPC_BIN_PREFIX)-objdump
+
+MWCC := tools/mwcc_compiler/$(MWCC_VERSION)
```
Comment (Thar0) [inline Makefile:50]:
```suggestion
MWCC_DIR := tools/mwcc_compiler/$(MWCC_VERSION)
```

File: tools/elf2dol/elf2dol.c [hunk 3]
```diff
@@ -0,0 +1,497 @@
+#include <stdlib.h>
```
Comment (Thar0) [inline tools/elf2dol/elf2dol.c:1]: Could you reformat this file to use spaces?

File: Makefile [hunk 4]
```diff
@@ -77,24 +92,36 @@ PROCFLAGS := -fprologue-fixup=old_stack
...
 
+distclean:
+	rm -f -r SIM.elf SIM_S.elf
+	$(MAKE) -C tools/elf2dol clean
+
 format:
 	clang-format -i include/*.h src/*.c
 
-.PHONY: all setup clean format
+# Note: this is meant for testing/modding purposes
```
Comment (Thar0) [inline Makefile:121]:
```suggestion
# Note: this is meant for testing/modding purposes as a dol is easier to package and run than the original elf
```
Or something, so a reader can know the motivation

**Review by Thar0 — CHANGES_REQUESTED** [review]
Some little things

**Review by Thar0 — APPROVED** [review]

**Pull request merged**